src/utils/helpers/createEvent/index.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Override a SyntheticEvent with a custom event.target
+ * This is intended to be used by Carbon components that provide a formatted value or those that need
+ * to provide a name or value that does no appear in the DOM e.g. a component that has two controls associated with one
+ * input.
+ */
+export default (event, overrides) => {
+  if (!event) {
+    throw new Error('Unable to extend event because event does not exist');
+  }
+
+  if (!overrides) {
+    throw new Error('Unable to extend event because no overrides were provided');
+  }
+
+  const { name = event.target.name, id = event.target.id, value = event.target.value } = overrides;
+
+  event.target = {
+    ...(value !== undefined && { value }),
+    ...(name && { name }),
+    ...(id && { id })
+  };
+
+  return event;
```

Comment:
I think that could be:
```
  return { target: { ...event.target, ...overrides } };
```
it will replace only defined values, or I'm missing something?

Reply:
We only want `name`, `id` and `value` but we don't want the properties defined in the object if the values are undefined.

Reply:
then we could just delete them:

```javascript
const target = { ...event.target, ...overrides };

Object.keys(target).forEach(key => {
  if (target[key] === undefined) {
    delete target[key];
  }
});

return { target };
```

Reply:
Deleting Object properties is still a slow operation to discourage this we explicitly said not to iterate over `event.target` in the RFC (so spreading `event.target` is prohibited). I'm going to dismiss this comment. If the solution becomes an issue we will revisit it at a later date.

Reply:
Deleting Object properties is still a slow operation to discourage this we explicitly said not to iterate over event.target in the RFC (so spreading event.target is prohibited). I'm going to dismiss this comment. If the solution becomes an issue we will revisit it at a later date